### ISSUE TITLE
`zola serve` panics when a new file is created or fails to parse toml

### ISSUE THREAD

**Issue by aslowwriter** [getzola/zola#1584]
# Bug Report
Currently `zola serve` will panic if you create an empty file. While working on my websites I often have `zola serve` active and create files from my editor, causing it to panic before I have a chance to add the front matter section. This is really quite annoying and now how other errors like accessing an out of scope variable in a template are handled. I would love it if zola could recover from failing to parse a front-matter section like it can for other errors. 

## Environment

Zola version: 0.13.0

## Expected Behavior
For example, if a template access a variable that is not in scope zola will not panic but simply abort rendering and display an error.  I'd expect the same to happen when an error is encountered during the processing of the front matter

## Current Behavior
Currently, this does not happen when a problem is encountered while parsing the front matter toml, or an empty file is parsed. when this happens zola panics and the program exits so it has to be restarted manually. 

## Step to reproduce
```bash
mkdir bug-report;
cd bug-report
zola init
zola serve & # the & is just so we can add the new file while serve is active
touch content/example.md
```

I'm guessing this isn't too difficult to implement. I'd imagine the `unwrap` etc. calls in the front matter section just need to be modified to hook into the same error handling code the other parts of zola use. I'm willing to see if I can make a PR for this if it would be likely to get merged.

**Comment by Keats** [comment]
I'm pretty sure it has been fixed in 0.14

**Comment by aslowwriter** [comment]
Sorry, I didn't know there was a new stable release yet. It has indeed been fixed in that one.
